newtypes.ts

```diff
@@ -0,0 +1,63 @@
+// Quicktype produces a definitions that are correct, but not as narrow or
+// well-named as hand-written type definition might produce. This module takes
+// the Quicktype-generated types as renames or modifies the types to be somewhat
+// nicer to work with in TypeScript.
+
+import type {
+  BaselineEnum as BaselineHighLow,
+  BrowserData,
+  Browsers,
+  GroupData,
+  FeatureData as QuicktypeMonolithicFeatureData,
+  WebFeaturesData as QuicktypeWebFeaturesData,
+  Release,
+  SnapshotData,
+  Status,
+  StatusHeadline,
+  Support,
+} from "./new.quicktype";
+
+// Passthrough types
+export type {
+  BaselineHighLow,
+  BrowserData,
+  Browsers,
+  GroupData,
+  Release,
+  SnapshotData,
+  Status,
+  StatusHeadline,
+  Support,
+};
+
+export interface WebFeaturesData
+  extends Pick<QuicktypeWebFeaturesData, "browsers" | "groups" | "snapshots"> {
+  features: { [key: string]: FeatureData };
+}
+
+export type FeatureData = Required<
+  Pick<
+    QuicktypeMonolithicFeatureData,
+    "description_html" | "description" | "name" | "spec" | "status"
+  >
+> &
+  Partial<
+    Pick<
+      QuicktypeMonolithicFeatureData,
+      "caniuse" | "compat_features" | "discouraged"
+    >
+  >;
```

Comment:
This is a preview of what's coming when we add redirects. TypeScript and JSON Schema's type systems aren't strictly equivalent, so quicktype will generate correct but overbroad types. In this module, I'll pluck out some nuances (and badly generated names) that quicktype doesn't handle very well and clean them up by _extending_ the generated types. While it's a bit weird looking, the benefit is that we can't completely diverge from the underlying JSON Schema.

Reply:
Do the TypeScript error messages if you get types wrong end up looking completely bizarre with this layering?